### DISCUSSION TITLE
2.1.3 keeping DAGs in scheduled state

### DISCUSSION THREAD

**Discussion by jakawell** [apache/airflow#17997] [Q&A]
We recently upgraded from 2.0 to 2.1.3 and we're now having an issue where whenever any more than a single DAG is running in parallel, every DAG seems to get stuck in the scheduled state with the "The scheduler is down or under heavy load" message. 

We utilize the `TriggerDagRunOperator` extensively, so we have a single DAG that triggers 3 DAGs in parallel, each of which triggers as many as 17-20 DAGs, though those 3 DAGs will only trigger up to 5 at once. We had no issues before upgrading to 2.1.3, so it seems to be something related to that upgrade. 

The pool as plenty of slots available (we have 128 slots and it's only using up to 17 or so in our debugging) and all the DAG tasks are being assigned to the proper pool.

Any ideas, at least of things to try as we debug?

**Comment by potiuk** [comment]
I think those **might** be fixed by some of the upcoming fixes in 2.1.4 . We had Not sure if any of the recent fixes are 100% relevant to your problem - it does not sound very familiar. Maybe you could share some more suspicious logs and some relevant extract of your DAGs so that we can take a look ?

**Reply by jakawell** [reply]
Thanks; yeah I'm planning to try to put together a more minimal example to reproduce what we're seeing. I'll get to this next week after the holiday.

**Reply by argibbs** [reply]
Having seen this myself a couple (ok, lots) of times that message has two common classes of cause:
1. Scheduler is not down, but db is busy. Check your (postgres?) db instance, if you can. Heartbeating is (AFAIK) done by tickling rows in the db, so if your instance is struggling, you'll see those messages (for us it's often caused by a really busy underlying filesystem)
2. Scheduler really _is_ down. Check your scheduler; if there's a poison pill dag (has happened to us in the past when I forgot to apply all the alembic updates) your scheduler might be in a fairly tight crash-restart loop.

Or of course, it might be something completely different.
